### PR TITLE
[Enhancement] Add listing url link to recruiting campaigns

### LINKED ISSUE #511: [Enhancement] Add listing url link to recruiting campaigns

### PR DESCRIPTION
**Pull request by JavaLavaMT**
closes #511 

image

### CODE REVIEW

File: app/components/list-item/job-opening-item.js [hunk 1]
```diff
@@ -1,3 +1,17 @@
 import BaseLiComponent from './base';
+import { computed } from '@ember/object';
+import { inject as service } from '@ember/service';
 
-export default BaseLiComponent.extend({ classNames: [ 'content' ] });
+export default BaseLiComponent.extend({
+  auth:       service(),
+  classNames: [ 'content' ],
+
+  quartzLink: computed('model.title', function () {
```
Comment (James1x0) [inline app/components/list-item/job-opening-item.js]: I'd rather you just eliminate this computed property by moving slug computation to the job-opening model like quartz has... https://github.com/associatedemployers/quartz/blob/master/app/models/job-opening.js#L38-L41

In the template you just use "myModel.slug"

File: app/components/list-item/job-opening-item.js [hunk 2]
```diff
@@ -1,3 +1,17 @@
...
+    if (this.model) {
+      let title = this.model.title + '';
+      return (title.match(/\s/) ? title.replace(/\s/, '-') : title)  + '_' +  this.model.id;
+    }
+  }),
+
+  companyPrefix: computed.reads('auth.user.company.urlPrefix')
```
Comment (James1x0) [inline app/components/list-item/job-opening-item.js]: Just link to this. You're only using it once.

File: app/templates/components/display/resource-list.hbs [hunk 3]
```diff
@@ -10,7 +10,9 @@
         {{component itemComponent item class=componentClass}}
       {{/link-to}}
     {{else}}
-      {{component itemComponent item class=componentClass}}
+      <div class={{linkClass}}>
```
Comment (James1x0) [inline app/templates/components/display/resource-list.hbs]: Just use componentClass in your declaration. Don't change this component.

File: app/templates/components/list-item/job-opening-item.hbs [hunk 4]
```diff
@@ -2,8 +2,13 @@
   {{yield model}}
 {{else}}
   <div class="header">
-    {{model.name}}
+    {{#link-to "account.job-opening" model.id}}
+      {{model.name}}
+    {{/link-to}}
   </div>
+  {{#if model.hiring}}
+    <span class="ui right floated"><a href="https://{{companyPrefix}}.granitehr/careers/{{quartzLink}}" target="_blank" rel="noopener">View careers link <i class="external alternate fitted icon"></i></a></span>
```
Comment (James1x0) [inline app/templates/components/list-item/job-opening-item.hbs]: This url is invalid AF

File: app/components/list-item/job-opening-item.js (hunk 2 above)
Comment (JavaLavaMT) [reply app/components/list-item/job-opening-item.js]: 🏁

File: app/components/list-item/job-opening-item.js (hunk 1 above)
Comment (JavaLavaMT) [reply app/components/list-item/job-opening-item.js]: nice! 🏁

File: app/components/list-item/job-opening-item.js (hunk 2 above)
Comment (JavaLavaMT) [reply app/components/list-item/job-opening-item.js]: 🏁

File: app/templates/components/display/resource-list.hbs (hunk 3 above)
Comment (JavaLavaMT) [reply app/templates/components/display/resource-list.hbs]: 🏁

File: app/templates/components/list-item/job-opening-item.hbs (hunk 4 above)
Comment (JavaLavaMT) [reply app/templates/components/list-item/job-opening-item.hbs]: yes it was, just keeping you on your toes :)

**Pull request merged**